comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process implements the local process collector for Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID   = "local-process-collector"
+	componentName = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+// NewCollector returns a new local process collector provider and an error
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:      collectorID,
+			catalog: workloadmeta.NodeAgent,
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) enabled() bool {
+	if flavor.GetFlavor() != flavor.DefaultAgent {
+		return false
+	}
+
+	processChecksInCoreAgent := config.Datadog().GetBool("process_config.process_collection.enabled") &&
+		config.Datadog().GetBool("process_config.run_in_core_agent.enabled")
+	langDetectionEnabled := config.Datadog().GetBool("language_detection.enabled")
+
+	return langDetectionEnabled && processChecksInCoreAgent
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	if !c.enabled() {
+		return errors.NewDisabled(componentName, "core agent language detection not enabled")
+	}
```

Comment:
IMHO, the error message here might be misleading. If language detection was enabled but live process collection / running in core agent is disabled, it will log that language detection is disabled, which is not the case. 

WDYT?

Reply:
Updated the error message to "language detection or core agent process collection is disabled". Let me know what you think.